# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/loader/URLPreloader.h
# Commit: aca7cd850cc9
# Full Hash: aca7cd850cc9c3a88dfcd82d58268929aa288497
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-08 09:42:17
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r d2e40366118c -r aca7cd850cc9 js/xpconnect/loader/URLPreloader.h
--- a/js/xpconnect/loader/URLPreloader.h	Wed Jul 08 02:46:11 2020 +0000
+++ b/js/xpconnect/loader/URLPreloader.h	Wed Jul 08 02:46:34 2020 +0000
@@ -26,8 +26,6 @@
 #include "nsIThread.h"
 #include "nsReadableUtils.h"
 
-class nsZipArchive;
-
 namespace mozilla {
 namespace loader {
 class InputBuffer;
@@ -77,7 +75,7 @@
   static Result<const nsCString, nsresult> ReadFile(nsIFile* file,
                                                     ReadType readType = Forget);
 
-  static Result<const nsCString, nsresult> ReadZip(nsZipArchive* archive,
+  static Result<const nsCString, nsresult> ReadZip(CacheAwareZipReader* archive,
                                                    const nsACString& path,
                                                    ReadType readType = Forget);
 
@@ -198,7 +196,7 @@
       return "";
     }
 
-    already_AddRefed<nsZipArchive> Archive() {
+    already_AddRefed<CacheAwareZipReader> Archive() {
       return Omnijar::GetReader(OmnijarType());
     }
 